src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,219 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Parses given string to operate on.
+ *
+ * @author Aaron Tan
+ */
+public class Parser {
+
+    private TaskList tasks;
+
+    /**
+     * Processes String s to be operated on.
+     *
+     * @param s String to be processed.
+     * @param tasks Task list to be operated on.
+     */
+    protected void process(String s, TaskList tasks) {
+        this.tasks = tasks;
+        String[] words = s.split(" ");
+        String command = words[0];
+        switch (command) {
+        case "list":
+            outputList();
+            break;
+        case "done":
+            markItemDone(s);
+            break;
+        case "unmark":
+            markItemUndone(s);
+            break;
+        case "todo":
+            insertTodo(s);
+            break;
+        case "deadline":
+            insertDeadline(s);
+            break;
+        case "event":
+            insertEvent(s);
+            break;
+        case "delete":
+            deleteTask(s);
+            break;
+        case "find":
+            find(s);
+            break;
+        default:
+            System.out.println("sorry, I don't understand you");
+            break;
+        }
+    }
+
+    /**
+     * Prints out tasks in list in format specified in each tasks' toString function.
+     */
+    protected void outputList() {
+        if (tasks.size() == 0) {
+            System.out.println("you got no tasks");
+        } else {
+            System.out.println("heres your tasks");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.format("\t%d.%s\n", i + 1, tasks.get(i));
+            }
+        }
+    }
+
+    /**
+     * Converts a string to a Todo and inserts into task list.
+     *
+     * @param input String to be converted to a Todo.
+     */
+    protected void insertTodo(String input) {
```

Comment:
Might I suggest that this method and others below it be abstracted out into their respective classes?